### ISSUE TITLE
How to plot full screen img or without scroll bar

### ISSUE THREAD

**Issue by kitaev-chen** [ioam/holoviews#2879]
My previous workflow is to write code with python and analyze/preview data in Matlab. Recently I want to go jupyter notebook with all of them so I explore holoviews in pyviz. It has fewer features compared with Matlab so I hope someone can let me know if it can be realized in holoviews. 

1. popup/fullscreen plot with interaction zoom. 
    - There is no fullscreen/popup image to see the detail of data
    - also, %matplotlib qt doesn't work
    - I use ```hv.extension('bokeh', 'matplotlib', width=100)``` so the width is bigger, but for big image, the height is limited within scroll bar. It's terrible to observe the detail 
![image](https://user-images.githubusercontent.com/6076946/42846577-58403e5a-89e7-11e8-816f-0c16d6af5fa4.png)

2. horizontal/vertical zoom with area selection by mouse.
    - I know how to zoom h or v (by wheel zoom) but mouse zoom with selection interval is convenient and precise.

3. annoying legend 
    - how to remove it?

Hope holoviews can be better choice than Matlab, for analyzing huge data!
Thanks.

**Comment by jbednar** [comment]
> .1. popup/fullscreen plot with interaction zoom.
> There is no fullscreen/popup image to see the detail of data

Yes, it would be great to have a "fullscreen" button on Bokeh plots that would expand them into full-screen mode like a video player does. This would need to be implemented by Bokeh, not HoloViews, and there is already a [Bokeh feature request for that](https://github.com/bokeh/bokeh/issues/3598
). That issue was opened in 2016, so it's obviously not a priority for the existing developers, but if you want to help it along you can try implementing it; that issue already links to some sample code.

> also, %matplotlib qt doesn't work

Right; I don't think anyone has ever worked on Qt support for the HoloViews matplotlib backend, and I don't expect anyone to do so in the foreseeable future.

> I use hv.extension('bokeh', 'matplotlib', width=100) so the width is bigger, but for big image, the height is limited within scroll bar. It's terrible to observe the detail

I'm not sure what you are hoping to happen here, but if it's just that you don't want the inner scrollbar, you can click on the area to the left of the plot, along the left margin, to disable the scrollbar and put the entire output onto the page.  Of course, you'll still need to use the whole page's scrollbar at that point if it's a very tall plot; not much that can be done about that!

> horizontal/vertical zoom with area selection by mouse.
> I know how to zoom h or v (by wheel zoom) but mouse zoom with selection interval is convenient and precise.

You can choose between Wheel Zoom or Box Zoom, as you prefer:
![image](https://user-images.githubusercontent.com/1695496/42848593-bef1c65c-89e5-11e8-9007-db586be7c1ee.png)

Box zoom sounds like what you want here.

> annoying legend  how to remove it?

You can use `.options(show_legend=False)` to turn off the legend, but then you'll have trouble figuring out which curve is which.  Or you can just move it to a different location, using `legend_position`.

**Comment by kitaev-chen** [comment]
@jbednar  very detail and many thanks!

for the "horizontal/vertical zoom with area selection by a mouse" part, sorry for my unclear description. In Matlab, you can limit h or v zoom only, by mouse selection with a <-------> mark. you don't need to find exact axis carefully every time. otherwise, you will mess up your image and there is no 'undo' button to recover your scaled image. 

Box Zoom can do precisely choose but you still need to adjust the height most of the time. Wheel Zoom can limit h or v on axis but you can not choose an interval precisely.

**Comment by ea42gh** [comment]
User guide Linking_Plots.ipynb shows an example using a RangeLink to achieve the zooming behavior I believe you are looking for.  It's also discussed in issue #2879

**Comment by jbednar** [comment]
@ea42gh, 2879 is *this* issue; did you mean to link to a different one?

**Comment by ea42gh** [comment]
@jbednar guess I am too eager to use all the really great features you all have been adding!
My version is strait from github master!

The issue I meant to reference is #2832

**Comment by kitaev-chen** [comment]
There are two new questions after using it for one day:
1.  how to set wheel_zoom as default used?
2.  hv.extension('bokeh', 'matplotlib', width=100) just set the notebook's width, how to set the width of hvplot image as full width?

**Comment by jbednar** [comment]
> @ea42gh: Would you please explain how I should think about master and release?

Sure. Glad you are enjoying the bleeding edge! Things that are in master are very likely to be in the next release, but the interface and functionality can change without notice.  So it's a good bet, but it's still a bet.  E.g. we very often rename classes, functions, and arguments once we've started using the new features and realize that something is confusing about them.  Once things have been released, we only change them for a very good reason, but until release we'll change even for a trivial amount of improvement.

But my point in mentioning master above was simply to alert @kitaev-chen that what you are mentioning is only in master, in case he looked for it in the release and got confused.

**Comment by jbednar** [comment]
@kitaev-chen, I don't think there is currently a way to set which tool should be active by default; see https://github.com/ioam/holoviews/issues/2888 .
